starter-code/src/memory.js

```diff
@@ -1,15 +1,36 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function(cards) {
+  this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
+};
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+MemoryGame.prototype.shuffleCard = function(cardsArr) {
+  //old school!!
+  for (i = 0; i < cardsArr.length; i++) {
+    var j = Math.floor(Math.random() * cardsArr.length);
+    var temp = cardsArr[i];
+    cardsArr[i] = cardsArr[j];
+    cardsArr[j] = temp;
+  }
+  return cardsArr;
+};
 
-// };
+MemoryGame.prototype.checkIfPair = function(firstCard, secondCard) {
+  this.pairsClicked++;
+  if (firstCard === secondCard) {
+    this.pairsGuessed++;
+    return true;
+  } else {
+    return false;
+  }
+};
 
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
-
-// }
-
-// MemoryGame.prototype.finished = function () {
-
-// };
+MemoryGame.prototype.finished = function() {
+  //if(this.pairsGuessed === (this.cards.length / 2)){ //THIS IS NOT WORKING WITH JASMINE
```

Comment:
It should work :sweat_smile: We teke note and will check the tests to find a possible error. Thanks for the report